doc/specs/#147 - Support different release channels [released, beta, alpha].md

```diff
@@ -0,0 +1,259 @@
+PS> winget search --channel "" # Return all packages which have a null channel
+PS> winget search <query> --channel <channel> # Return packages matching <query> which have a channel matching <channel>
+PS> winget search --tag <tag> --channel <channel> # Return packages with a tag matching <tag> which have a channel matching <channel>
+...
+```
+
+#### winget show
+When the user specifies `--versions`, the versions for all channels along with the channels should be shown. The versions should be grouped by channel, sorted by channel, then sorted by version descending
+When the user specifies `--channels`, the available channels for the package should be shown. If a package contains a named default channel and unnamed (null) channels, only the named channels should be shown
+
+#### winget install
+When the user does not specify `--channel`, the default channel should be used. If the user does specify `--channel`, the specified channel should be used. The latest version for the channel should be installed, unless a user specified the version using the `--version` parameter. Additionally, the Windows Package Manager should record the package channel in some manner to ensure that upgrades to the package are found using the correct channel.
+
+Optionally, the user should be able to specify the channel as part of the package identifier using the `@` symbol. For example, `winget install Google.Chrome@dev` would be equivalent to `winget install Google.Chrome --channel dev`
+
+#### winget upgrade
+When listing the packages available for upgrade, the Windows Package Manger should look only for packages matching the release channel of the installed package. 
+When installing packages available for upgrade, the Windows Package Manager should only install new versions matching the release channel of the installed package.
+
+If the package channel cannot be determined, there are several options. 
+1) Ask the user for input.
+2) Throw a warning and skip that package
+3) Use the default channel for that package
+
+The Windows Package Manager should never leave a package in a state where it cannot be upgraded simply because the package is unknown. To account for this, there must be some way for the user to specify a release channel. Therefore, when `winget upgrade --all` is used, and the release channel cannot be determined, a warning should be shown and the package should be skipped. For all other cases, the user should be asked to select a release channel on a per-package basis; the user's choice for that package should persist.
+
+#### winget export/import
+When exporting the list of installed packages, where possible, the release channel should be included with the export.
+When importing the list of installed packages, where possible, the release channel should be honored. If the release channel is not able to be honored, a warning should be thrown, and the default release channel should be used
```

Comment:
I think our default stance is to fail the import without doing anything if we know up front that we can't do what is asked.

Reply:
Thats a fair stance. I’ll update this if needed